### PR TITLE
Spp 10021 stop marker and input types

### PR DESCRIPTION
**Pull request by JasonBellONS**
## Synopsis

Thousand pounds updated to now take a dictionary containing the target variables values then creating the objects within the method,  TpcMarker enum created and Check_zero_errors implemented to check for Stop scenarios. Tests and pandas wrapper updated

## Checklist

- [x] Documentation created/updated
- [x] Tests created/updated

## Description

NOTE: PR RELIES ON #53 being merged first.

### CODE REVIEW

File: sml_small/editing/thousand_pounds/thousand_pounds.py [hunk 1]
```diff
@@ -29,6 +29,17 @@ class InputParameters(Enum):
     TARGET_VARIABLES = 5
 
 
+class TpcMarker(Enum):
+    """
+    Enum for use when setting/comparing tcc_marker values
```
Comment (gibbardsteve) [inline sml_small/editing/thousand_pounds/thousand_pounds.py:34]: comment mentions tcc_marker , tpc

File: sml_small/editing/thousand_pounds/thousand_pounds.py [hunk 2]
```diff
@@ -69,7 +80,7 @@ def thousand_pounds(
     auxiliary: Optional[float],  #
     upper_limit: float,  #
     lower_limit: float,  #
-    target_variables: List[Target_variable],
+    target_variables: dict,
 ) -> Thousands_output:
     """
     Calculates a pounds thousands error ratio and if the ration is between the bounds of the given limits will adjust
```
Comment (gibbardsteve) [inline sml_small/editing/thousand_pounds/thousand_pounds.py:86]: ration should be ratio...

File: sml_small/editing/thousand_pounds/thousand_pounds.py [hunk 3]
```diff
@@ -267,17 +299,47 @@ def validate_input(
...
+    :param predictive: Value used for 'previous' response (Returned/Imputed/Constructed)
+    :type predictive: Optional[float]
+    :param auxiliary: Calculated response for the 'previous' period
+    :type auxiliary: Optional[float]
+
+    :return: tpc_marker, either method_proceed or stop
+    :rtype: TpcMarker
+    """
+    tpc_marker = TpcMarker.METHOD_PROCEED
+    if predictive == 0 and (auxiliary is None or auxiliary == 0):
```
Comment (gibbardsteve) [inline sml_small/editing/thousand_pounds/thousand_pounds.py:318]: What happens if predictive is None and auxiliary is 0?

**Review by gibbardsteve — COMMENTED** [review]
Did the testing work out ok and have you had a look through ELlie's UAT test doc to see if what we have is largely following her expectations?

File: sml_small/editing/thousand_pounds/thousand_pounds.py (hunk 1 above)
Comment (JasonBellONS) [reply sml_small/editing/thousand_pounds/thousand_pounds.py]: amended

File: sml_small/editing/thousand_pounds/thousand_pounds.py (hunk 2 above)
Comment (JasonBellONS) [reply sml_small/editing/thousand_pounds/thousand_pounds.py]: amended

File: sml_small/editing/thousand_pounds/thousand_pounds.py (hunk 3 above)
Comment (JasonBellONS) [reply sml_small/editing/thousand_pounds/thousand_pounds.py]: amended to also set stop if predictive is None, spoke to Ellie and she agrees that it should be a scenario that outputs a stop value and will update the test data next week

File: sml_small/editing/thousand_pounds/thousand_pounds.py [hunk 4]
```diff
@@ -267,17 +299,47 @@ def validate_input(
...
+        tpc_marker = TpcMarker.STOP
```
[the remark this answers is not in the document]
Comment (JasonBellONS) [reply sml_small/editing/thousand_pounds/thousand_pounds.py:331]: amended within main method

**Review by gibbardsteve — APPROVED** [review]

**Pull request merged**